sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala

```diff
@@ -173,7 +174,8 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog {
 
           case s: Star =>
             withPosition(s) {
-              throw QueryCompilationErrors.invalidStarUsageError(operator.nodeName)
+              throw QueryCompilationErrors.invalidStarUsageError(operator.nodeName,
+                conf.supportQuotedRegexColumnName)
```

Comment:
Is this correct? Even if `supportQuotedRegexColumnName` is true,  a `Star` expression is not necessarily a regexp expression.